lib/phobos.rb

```diff
@@ -31,19 +31,44 @@
 
 Thread.abort_on_exception = true
 
+# Monkey patch to fix this issue: https://github.com/zendesk/ruby-kafka/pull/732
+module Logging
+  # :nodoc:
+  class Logger
+    # :nodoc:
+    def formatter=(*args); end
+
+    # :nodoc:
+    def push_tags(*args); end
+
+    # :nodoc:
+    def pop_tags(*args); end
+  end
+end
+
 module Phobos
   class << self
     attr_reader :config, :logger
     attr_accessor :silence_log
 
     def configure(configuration)
       @config = fetch_configuration(configuration)
-      @config.class.send(:define_method, :producer_hash) { Phobos.config.producer&.to_hash }
-      @config.class.send(:define_method, :consumer_hash) { Phobos.config.consumer&.to_hash }
+      config_define_methods
       @config.listeners ||= []
       configure_logger
     end
 
+    def config_define_methods # rubocop:disable Metrics/AbcSize
```

Comment:
```
rubocop:disable Metrics/AbcSize
```

Is a smell and the naked eye can see that this method is very complex (which is what rubocop tries to find out for you) and should be avoided if possible.

Reply:
Yeah... it was off by 0.3 and tbh I think the default settings for Rubocop are a bit too extreme. There's always a continuum of what people consider "complex". :) In this case I felt the function was simple enough to override the minor infraction, but given the changes I'm thinking of doing, this might be moot.

Reply:
I think with the other changes/actions suggested the ABC size can be reduced so this could be removed